docs/technical.rst

```diff
@@ -95,8 +95,26 @@ Features supported, encouraged and discouraged from the different ``JavaScript``
 TypeScript
 ^^^^^^^^^^
 
-``TODO``
+TypeScript is a type-safe version of JavaScript that compiles down to JavaScript. Type-safety
+offers a better development experience and improves the reliability and maintainability of
+large projects. TypeScript also produces type definition files to allow upstream projects to
+safely use EthereumJS libraries.
+
+TypeScript language support is a compile-time feature separate from the runtime JavaScript support.
+All EthereumJS projects that are written in TypeScript are compiled using TypeScript 3.2 or newer.
+
+Notable language features:
+
+- Async Iteration
+
+- `Generics <https://www.typescriptlang.org/docs/handbook/generics.html>`_
+
+- `Function Overload <https://www.typescriptlang.org/docs/handbook/functions.html>`_
```

Comment:
I find this too generic. These language features can be mentioned but the main point from the docs here should be to state what concrete language features from the concrete ``TypeScript`` versions we agree on to use in the projects.

Can you add some structure (doesn't need to be complicated) that allows to e.g. once ``3.3`` comes out add two bullet points or so on what features from that version we are intending to use and which maybe not yet.

Maybe let's start with a (can be really short, just to have the structure in place) ``3.2`` section and do 1-2 statements about interesting language features (we don't have to go down the whole list, that would definitely overblow the documentation and is not realistically to be maintained).

I think the most interesting part is the ``BigInt`` introduction, maybe we can start with that, since we make heavy use of big integers - currently in the form of the [bn.js](https://github.com/indutny/bn.js/) library - in our repos, especially the virtual machine.

My personal current stance on this after a first look is: this is super-interesting, but we can't use it yet since it is compiling to an experimental target. If you would agree here, we can start dropping a note on the README about this.

Optimally also link to the ``TypeScript`` ``3.2`` feature page along the line.

Reply:
> I find this too generic. These language features can be mentioned but the main point from the docs here should be to state what concrete language features from the concrete TypeScript versions we agree on to use in the projects.

> Can you add some structure [that allows once newer versions come out to say which features] we are intending to use and which maybe not yet.

Unlike JavaScript, there aren't many (new) warts in the language. There aren't many (any?) features that we _shouldn't_ be using as a general statement.

> I think the most interesting part is the `BigInt` introduction, maybe we can start with that, since we make heavy use of big integers - currently in the form of the bn.js library - in our repos, especially the virtual machine.

I agree that any experimental features shouldn't yet be used in stable libraries, I noted that below but I can make that more explicit.